daemon/container_linux.go

```diff
@@ -306,6 +307,25 @@ func populateCommand(c *Container, env []string) error {
 	return nil
 }
 
+func mergeDevices(defaultDevices, userDevices []*configs.Device) []*configs.Device {
+	if len(userDevices) == 0 {
+		return defaultDevices
+	}
+
+	paths := map[string]*configs.Device{}
+	for _, d := range userDevices {
+		paths[d.Path] = d
+	}
+
+	var devs []*configs.Device
+	for _, d := range defaultDevices {
+		if _, defined := paths[d.Path]; !defined {
+			devs = append(devs, d)
+		}
+	}
+	return append(devs, userDevices...)
```

Comment:
I think that the current code is clear and correct. Did it occure to you to do it the other way around and write first the default devices into the map and then overwrite the map using the userDevices?

````
paths := map[string]*configs.Device{}
// First add the default devices to the path map.
for _, d := range defaultDevices {
 paths[d.Path] = d
}
//Then over-ride those defaults when necessary.
for _, d := range userDevices {
 paths[d.Path] = d
}
var devs []*configs.Device
for _, d := range paths {
  devs = append(devs,d)
}
return devs
````

I'm actually not sure if this works in golang, it's the pythonic way of doing this though.